### PR TITLE
Most accepted fix to red_ribbon_arm so far

### LINKED ISSUE #16443: Arm sash
(WEB REPORT BY: repertor REMOTE: 172.93.109.202:7777)
# Revision
1440a12fe5a71682511314726522fddf280c4608

# Description
Quick swapping the sash will give you a permanent arm
# Steps to Reproduce
quickswap sash on belt
# What you Expected
remove arm
# What Actually Happened
multiple arms

### PR DESCRIPTION
**Pull request by ghost**
Fixes #16443

### CODE REVIEW

File: code/game/objects/items/misc_items.dm [hunk 1]
```diff
@@ -49,7 +49,7 @@
 
 /obj/item/red_ribbon_arm/equipped(mob/living/carbon/human/H, equipped_slot)
 	..()
-	if(istype(H) && H.get_item_by_slot(slot_belt) == src)
+	if(istype(H) && H.get_item_by_slot(slot_belt) == src && equipped_slot != null)
```
Comment (Shadowmech88) [inline code/game/objects/items/misc_items.dm:52]: Did you test this? Does this actually work?

The problem is that the code in `unequipped()`, which removes the user's extra hand, is not being executed when the item is unequipped via quick-swapping something else into the slot, allowing the user to keep the extra hand.
I'm not sure how the addition of another check into its `equipped()` proc would fix that.

**Comment by ghost** [comment]
tested

**Comment by Exxion** [comment]
So the the issue wasn't it failing to remove the hand, but instead was it incorrectly adding it back right afterwards.
(Or right before, but the order isn't important)

**Comment by Shadowmech88** [comment]
Interesting that `equipped()` would be called on the hand before the reference to the item was removed from the belt slot.

**Review by Shadowmech88 — APPROVED** [review]

**Pull request merged**